deeppavlov/dataset_readers/odqa_reader.py

```diff
@@ -0,0 +1,212 @@
+# Copyright 2017 Neural Networks and Deep Learning lab, MIPT
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import logging
+from pathlib import Path
+import unicodedata
+import sqlite3
+from typing import Union, List, Tuple, Generator, Any, Optional
+from multiprocessing import Pool
+
+from tqdm import tqdm
+
+from deeppavlov.core.data.dataset_reader import DatasetReader
+from deeppavlov.core.common.registry import register
+from deeppavlov.core.commands.utils import expand_path
+from deeppavlov.core.data.utils import download
+
+logger = logging.getLogger(__name__)
+
+
+@register('odqa_reader')
+class ODQADataReader(DatasetReader):
+    """Build a SQLite database from folder with txt files, json files or
+    `Wiki Extractor <https://github.com/attardi/wikiextractor>`_ files.
+
+    """
+
+    def read(self, data_path: Union[Path, str], db_url: Optional[str] = None, *args, **kwargs) -> None:
+        """Build a SQLite database from provided files, download SQLite database from a provided URL, or do nothing.
+
+        Args:
+            data_path: a directory/file with texts to create a database from
+            db_url: path to a database url
+
+        Returns:
+            None
+
+        """
+        logger.info('Reading files...')
+        save_path = expand_path(kwargs['save_path'])
+        if save_path.exists() and Path(save_path.parent / '.done').exists():
+            return
+
+        if db_url:
+            if Path(save_path).is_file():
+                download_dir = save_path.parent
+            else:
+                download_dir = save_path
```

Comment:
`Path(save_path).is_file()` will return `True` only if a file already exists. So it should not determine download directory

Reply:
ok